docusaurus/src/components/VersionChanger.tsx

```diff
@@ -0,0 +1,165 @@
+      sdkPage: null,
+    };
+  } else {
+    const sdkWordIdx = splitPath.findIndex((value) => value === "sdk");
+    const sdkInstance = splitPath[sdkWordIdx + 1];
+    const sdkPage = splitPath.slice(sdkWordIdx + 2).join("/");
+    const sdkVersion = isNaN(splitPath[sdkWordIdx + 2][0])
+      ? "current"
+      : splitPath[sdkWordIdx + 2];
+    return {
+      sdkInstance,
+      sdkPage,
+      sdkVersion,
+    };
+  }
+};
+
+const pathExists = (path, data) => {
+  return data.docs.some((doc) => doc.path === path);
+};
+
+function SDKSelector({ sdkSitesMetaData, activeSDK }) {
+  const history = useHistory();
+  function changeHandler({ value: sdkName }) {
+    const sdkMetadata = sdkSitesMetaData[sdkName];
+    const sdkBasePage = sdkMetadata.path + "intro"; //TODO: refactor into smthn better
+    history.push(sdkBasePage);
+  }
+  delete sdkSitesMetaData.default;
+  const options = Object.keys(sdkSitesMetaData).map((sdkId) => ({
+    value: sdkId,
+    label: getContext(sdkId).name,
+  }));
+  return (
+    <div style={{ padding: 5 }}>
+      <Select
+        options={options}
+        onChange={changeHandler}
+        styles={colorStyles}
+        defaultValue={options.find((option) => option.value === activeSDK)}
+      />
+    </div>
+  );
+}
+
+function ActiveSDKVersionSelector({
+  activeSdk,
+  activeSdkVersions,
+  currentSdkVersion,
+  sdkSitesMetaData,
+}) {
+  const history = useHistory();
+  function changeHandler({ value: sdkVersion }) {
+    const sdkMetadata = sdkSitesMetaData[activeSdk];
+    const sdkVersionBasePage =
+      sdkVersion === "current"
+        ? sdkMetadata.path + "intro"
+        : sdkMetadata.path + sdkVersion + "/intro"; //TODO: refactor into smthn better
+    history.push(sdkVersionBasePage);
+  }
+  const options = activeSdkVersions.map((version) => ({
+    value: version.name,
+    label: version.label,
+  }));
+  return (
+    <div style={{ padding: 5 }}>
+      <Select
+        options={options}
+        onChange={changeHandler}
+        styles={colorStyles}
+        defaultValue={options.find(
+          (option) => option.value === currentSdkVersion
+        )}
+      />
+    </div>
+  );
+}
+
+const ContextSwitcher = ({ className }) => {
+  const [context, setContext] = useState(CONTEXTS[0]);
+  const [versions, setVersions] = useState(null);
+  const data = useAllPluginInstancesData("docusaurus-plugin-content-docs");
+  const history = useHistory();
+  // TODO: refactor to make worth with this hook
+  // const activeDoc = useActiveDocContext(
+  //   getCurrentSiteInstance(history).sdkInstance
+  // );
+  // console.log({ activeDoc });
+
+  useEffect(() => {
+    const { sdkPage } = getCurrentSiteInstance(history);
+
+    const currContext = getContext(sdkPage);
+    if (currContext && currContext.id !== context.id) {
+      setContext(currContext);
+    }
+    const { sdkInstance: sdkName } = getCurrentSiteInstance(history);
+    setVersions(data[sdkName]?.versions);
+  }, []);
```

Comment:
Shouldn't "history" be in the deps list?
```suggestion
  }, [history]);
```